fabric/src/main/java/net/minescript/common/Minescript.java

```diff
@@ -3981,6 +4010,116 @@ private static Optional<String> handleScriptFunction(
         }
         return Optional.of(toJsonString(getScreenName().orElse(null)));
 
+      case "container_get_item": // Item in slot in container
+      {
+        if (args.size() != 1) {
+          logUserError("Error: `{}` expected 1 param (slot id) but got: {}", functionName, argsString);
+          return Optional.empty();
+        }
+        Screen screen = minecraft.currentScreen;
+        int slot = getStrictIntValue(args.get(0)).getAsInt();
+        if (screen == null) {
+          logUserError("Error: `{}`: slot {} does not exist", functionName, slot);
+          return Optional.empty();
+        }
+        if (screen instanceof HandledScreen<?>) {
+          HandledScreen handledScreen = (HandledScreen) screen;
+          ScreenHandler screenHandler = handledScreen.getScreenHandler();
+          if (screenHandler.slots.size() <= slot) {
+            return Optional.empty();
+          }
+          ItemStack itemStack = screenHandler.slots.get(slot).getStack();
+          return Optional.of(itemStackToJsonString(itemStack, OptionalInt.of(slot), false));
+        } else {
+          return Optional.empty();
+        }
+      }
+
+      case "container_get_items": // List of Items in Chest
+      {
+        Screen screen = minecraft.currentScreen;
+        if (screen instanceof HandledScreen<?>) {
+          HandledScreen handledScreen = (HandledScreen) screen;
+          ScreenHandler screenHandler = handledScreen.getScreenHandler();
+          Slot[] items = screenHandler.slots.toArray(new Slot[0]);
+          StringBuilder resultString = new StringBuilder("[");
+          for (int i = 0; i < items.length; i++) {
+            ItemStack itemStack = items[i].getStack();
+            if (itemStack.isEmpty()) {
+              continue;
+            }
+            resultString.append(itemStackToJsonString(itemStack, OptionalInt.of(items[i].id), false));
+            if (i < items.length - 1) {
+              resultString.append(",");
+            }
+          }
+          resultString.append("]");
+          return Optional.of(resultString.toString());
+        } else {
+          return Optional.empty();
```

Comment:
Returning `Optional.empty()` should be reserved for long-running script functions that can return multiple values like registration of an event listener. In this case, calling `container_get_items()` when there's no `HandledScreen<?>` GUI should either return `Optional.of("null")` or throw `IllegalStateException`.